### PR TITLE
Dact 385 list of active directors

### PR DESCRIPTION
**Pull request by KALatchman**
The current directors page is the 6th page in the happy path journey for the web service. It acts as a page where an Officer may select from a list of active directors associated with a company to remove. 

An Officer should be able to click the remove director link which will take them to the ‘When was this director removed ?’ (removal date) page.

To complete this work, we first need an endpoint to retrieve the list of active Directors.

image

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/officerfiling/api/controller/OfficerControllerImpl.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    private final Logger logger;
+
+    public OfficerControllerImpl(final OfficerService officerService, final Logger logger) {
+        this.officerService = officerService;
+        this.logger = logger;
+    }
+
+    @Override
+    @ResponseBody
+    @GetMapping(value = "/transactions/{transactionId}/officers/active-officers-details", produces = {"application/json"})
```
Comment (jbestwickch) [inline src/main/java/uk/gov/companieshouse/officerfiling/api/controller/OfficerControllerImpl.java:32]: Should this be marked as a private endpoint? I don't imagine third party filers would need to access it.

**Review by jbestwickch — COMMENTED** [review]
Comment regarding whether endpoint should be marked as private

File: src/main/java/uk/gov/companieshouse/officerfiling/api/controller/OfficerControllerImpl.java (hunk 1 above)
Comment (jbestwickch) [reply src/main/java/uk/gov/companieshouse/officerfiling/api/controller/OfficerControllerImpl.java:32]: Also we'd need to make sure it's protected from external access. Marking it as private will require an internal CH key to be used for auth.

**Comment by PhilipGlover** [comment]
This is probably more of an architectural decision, but there is a CHS endpoint for this already:

**Review by rwalrondch — DISMISSED** [review]

**Comment by rwalrondch** [comment]
> This is probably more of an architectural decision, but there is a CHS endpoint for this already: `<host>/company/00109065/officers`

Yes there is and we do use it elsewhere, But we were given a "Requirement" to show Full DOB on the active directors page,  which that endpoint doesn't give you, as it only shows month and year.

**Comment by PhilipGlover** [comment]
Ok, but cant the officer web service not call the Appointments API with the privilege to see the full data ie full date of birth?

**Comment by rwalrondch** [comment]
> Ok, but can't the officer web service not call the Appointments API with the privilege to see the full data ie full date of birth?

Yes you could,  but you would have to make the call per director, so not so bad for a 1 or 2 officer company,  but if you had 20 odd directors, could start getting really bad performance.

I will check this with Mike when he is back, but I think we should go with it for the moment?

**Pull request closed without merging**